JobScheduler.js

```diff
@@ -172,9 +174,25 @@ class JobScheduler {
                 .catch(err => reject('error occurred while fetching service fabrik broker status:', err));
             } else {
               logger.info('+-> System is in maintenance :', _.pick(maintenanceInfo, maintInfoAttrs));
+              const downTimePhase = maintenanceManager.getDowntimePhase(maintenanceInfo, config.scheduler.downtime_maintenance_phases);
+              if (downTimePhase === undefined) {
+                logger.info('Current Maintenance phase does not affect scheduler downtime. So going ahead and starting it...');
+                clearInterval(this.intervalTimer);
+                return resolve();
+              }
+              logger.info(`Scheduler in downtime phase: ${downTimePhase}, will wait till maintenance state changes from in-progress to completed state`);
+              const phaseStartTimeInMs = Date.parse(downTimePhase.substring(downTimePhase.lastIndexOf('at') + 2));
```

Comment:
instead of parsing this way , cant we get the start time of this phase depending on the `updatedAt` field of `maintenanceInfo` ?

Reply:
Actually thats what i wanted to do earlier, but the phases keep on getting updated. ex., SF Update, Registration update and then mongodb update. Now if i take the updated time into consideration, then it will be time of the last phase rather than the start of downtime phase. But yeah use updated time incase parsing of time fails.

Reply:
Actually made it work with updatedAt field itself :-)

Reply:
Ok , fine.